src/config-client/config-client.ts

```diff
@@ -8,17 +8,22 @@ import type {
   PropertySource,
 } from './config-client.interface';
 
-export function getConfigSync({
-  endpoint = 'http://localhost:8888',
-  application = 'application',
-  profile = 'default',
-  label = 'main',
-}: ClientRequestOptions): Config {
-  const url = `${endpoint}/${application}/${profile}/${label}`;
+const DEFAULT_CLIENT_REQUEST_OPTIONS = {
+  endpoint: 'http://localhost:8888',
+  application: 'application',
+  profile: 'default',
+  label: 'main',
+};
+
+export function getConfigSync(options: Partial<ClientRequestOptions> = DEFAULT_CLIENT_REQUEST_OPTIONS): Config {
+  const url = `${options.endpoint}/${options.application}/${options.profile}/${options.label}`;
```

Comment:
1. endpoint는 과연 저 값이 유의미한 default 값인지 모르겠습니다. 제 생각엔 default를 없애고 필수 속성으로 바꿔야할 것 같습니다
2. 이렇게 하면 아래 두 호출의 내용이 달라집니다.
```typescript
console.log(getConfigSync());
console.log(getConfigSync({}));
```

이 경우는 options 자체가 undefined이어야 할 필요는 없으므로 실장님 코드 혹은 아래 코드처럼 해야합니다.
```suggestion
export function getConfigSync(options: Partial<ClientRequestOptions>): Config {
  const { endpoint, application = 'application', profile = 'default', label = 'main' } = options;
  const url = `${endpoint}/${application}/${profile}/${label}`;
```

Reply:
말씀해주신대로 endpoint default value 제거하였고

`ClientRequestOptions` 의 인터페이스 멤버중
spring cloud config의 필수 요구값인 endpoint, application을 제외한 profile, label 만을 optional로 변경했습니다.
(Partial <> 제너릭 삭제)